### PR TITLE
bugfix:throw exception when update pk 

### PR DESCRIPTION
**Pull request by 81519434**
### Ⅰ. Describe what this PR did
optimize this pr. #3129

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java [hunk 1]
```diff
@@ -35,10 +24,19 @@
 import io.seata.rm.datasource.sql.struct.TableRecords;
 import io.seata.sqlparser.SQLRecognizer;
 import io.seata.sqlparser.util.JdbcConstants;
-import org.apache.commons.lang.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.sql.SQLException;
```
Comment (a364176773) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java:30]: java packages should be at the top

**Review by a364176773 — CHANGES_REQUESTED** [review]
please use p3C to format the code

**Comment by caohdgege** [comment]
关联一下#3289

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java (hunk 1 above)
Comment (81519434) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java]: done

**Comment by xingfudeshi** [comment]
This problem is that the primary key should not be changed. If the new primary key value is consistent with the old primary key value, can it be ignored directly?@81519434

**Comment by 81519434** [comment]
> This problem is that the primary key should not be changed. If the new primary key value is consistent with the old primary key value, can it be ignored directly?@81519434

Yes, this pr is doing exactly what you said!

**Review by xingfudeshi — APPROVED** [review]
LGTM.

**Review by slievrly — APPROVED** [review]
LGTM

**Review by a364176773 — APPROVED** [review]
LGTM

**Pull request merged**